test/e2e/components/vl-select.js

```diff
@@ -0,0 +1,112 @@
+const { VlElement } = require('vl-ui-core');
+const { By, Key } = require('selenium-webdriver');
+
+class VlSelect extends VlElement {  
+    async _getDressedContainer() {
+        return this.findElement(By.xpath('../..'));
+    }
+    
+    async _getSelectList() {
+        return (await this._getDressedContainer()).findElement(By.css('.vl-select__list'));
+    }
+    
+    async _getInput() {
+        return (await this._getSelectList()).findElement(By.css('.vl-input-field'));
+    }
+    
+    async _openDressedDropdown() {
+        return (await this._getDressedContainer()).click();
+    }
+    
+    async _getSelectParent() {
+        return this.findElement(By.xpath('..'));
+    }
+    
+    async _getDeleteButton() {
+        const parent = await this._getSelectParent();
+        return parent.findElement(By.css('button.vl-pill__close'));
+    }
+    
+    async _getOptions() {
+        if (await this.isDressed()) {
+            const selectList = await this._getSelectList();
+            return selectList.findElements(By.css('.vl-select__item'));
+        } else {
+            return this.findElements(By.css('option'));
+        }
+    }
+    
+    async _mapDressedOptions(options) {
+        return Promise.all(options.map(async (selectItem) => {
+            const text = await selectItem.getText();
+            return {text: text, webElement: selectItem};
+        }));
+    }
+                            
+    async isDressed() {
+        return this.hasAttribute('data-vl-select-dressed');
+    }
+    
+    async values() {
+        const options = await this._getOptions();
+        return Promise.all(options.map(o => o.getAttribute('textContent')));
+    }
+
+    async hasValue(value) {
+        const values = (await this.values()).map(v => v.trim());
+        return values.includes(value);
+    }
+
+    async selectValue(value) {
+        if(await !this.hasValue(value)) {
+            return Promise.reject('Value ' + value + ' niet gevonden in select!');
+        }
+        if((await this.isDressed())) {
+            await this._openDressedDropdown();
+            const selectItems = await this._getOptions();
+            const map = await this._mapDressedOptions(selectItems);
+
+            const webElementArray = map.filter(i => i.text == value);
```

Comment:
heet de methode dan niet beter selectByText?
maak je niet beter een API met verschillende manieren om te selecteren? selectByValue, selectByText, selectByIndex, ...

Reply:
k weet niet op je dan een "VlSelectOption" nodig hebt om value + text voor te stellen? en evt op te vragen of een bepaalde optie al dan niet disabled is, als dat er al in zit?

Reply:
+1 voor de selectByValue en selectByText etc. Dat pas ik nog aan.
Wat betreft de VlSelectOption, dat is mss idd een optie om dat wat te abstraheren, ik bekijk het.